index.js

```diff
@@ -1,117 +1,142 @@
-  
+
   if (!newMovie.name) {
-  const message = 'missing name in request body';
-  res.status(400).send(message);
+    const message = "missing name in request body";
+    res.status(400).send(message);
   } else {
-  newMovie.id = uuid.v4();
-  movies.push(newMovie);
-  res.status(201).send(newMovie);
-   }
-  });
-  
-  // deletes a movie from our list by ID \\
-  app.delete('/movies/:id', (req, res) => {
+    newMovie.id = uuid.v4();
+    movies.push(newMovie);
+    res.status(201).send(newMovie);
+  }
+});
+
+// deletes a movie from our list by ID \\
+app.delete("/movies/:id", (req, res) => {
   let movie = topMovies.find((movie) => {
-  return movie.id === req.params.id
+    return movie.id === req.params.id;
   });
-  
+
   if (movie) {
-  movie = movie.filter((obj) => {
-  return obj.id !== req.params.id 
-  });
-  res.status(201).send('movie ' + req.params.id + ' was deleted');
-  
-  }
-  });
-  
-  // make the "user" of a user \\
-  
-  app.put('/movies/:director/:genre/:imageURL/:user', (req, res) => {
-  let user = user.find((user) => {
-  return user.user === req.params.user
-  });
-  
-  if (user) {
-  user.user[req.params.user] = parseInt(req.params.user);
-  res.status(201).send('user ' + req.params.user + ' was added ' + req.params.grade + ' in ' + req.params.movies);
+    topMovies = topMovies.filter((obj) => {
+      return obj.id !== req.params.id;
+    });
+    res.status(201).send("movie " + req.params.id + " was deleted");
   } else {
-  res.status(404).send('user with the name ' + req.params.user + ' cannot be made.');
+    res.status(404).send("movie not found");
   }
-  });
+});
 
-// GET requests \\
-app.get('/', (req, res) => {
-res.send('Welcome to my app');
+//adds movie to the list of favorites of a user
+app.put("/users/:id/movies/:movie_id", (req, res) => {
+  res.send("Successful PUT request added movies to the list of favorites");
 });
 
-app.get('/secreturl', (req, res) => {
-res.send('this is a secret url with top secret content!')
+//removes movie from the list of favorites of a user
+app.delete("/users/:id/movies/:movie_id", (req, res) => {
+  res.send("Successful DELETE request added movies to the list of favorites");
 });
 
-app.get('/movies', (req, res) => {
-res.json(topMovies);
+app.post("/users", (req, res) => {
+  res.send("Successful POST request to register new user");
+});
+
+app.delete("/users", (req, res) => {
+  res.send("Successful DELETE request to register new user");
+});
+
+// gets the data about a single director, by name \\
+app.get("/directors/:name", (req, res) => {
+  res.send(
+    "Successful GET request to get data about a single director by name"
+  );
+});
+
+app.get("/genre/:name", (req, res) => {
+  res.send("Successful GET request to get data about a single genre by name");
+});
+
+// GET requests \\
+app.get("/", (req, res) => {
+  res.send("Welcome to my app");
+});
+
+app.get("/secreturl", (req, res) => {
+  res.send("this is a secret url with top secret content!");
 });
```

Comment:
The logic of these endpoints doesn't need to be implemented yet as you'll be working with a Mongo database to actually save the data so you don't have to worry about adding additional arrays for users, directors, or genres.